packages/code-du-travail-frontend/src/outils/DureePreavisDemission/steps/InfosCCn.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect } from "react";
+import { Field } from "react-final-form";
+import styled from "styled-components";
+import createPersistedState from "use-persisted-state";
+import { Button, Toast, theme } from "@socialgouv/react-ui";
+
+import { QuestionLabel } from "../../common/stepStyles";
+import Search from "../../../conventions/Search/Form";
+import { isNotYetProcessed } from "./situation";
+
+// store selected convention in localStorage
+const useConventionState = createPersistedState("convention");
+
+function StepInfoCCn({ form }) {
+  const [ccInfo, setCcInfo] = useConventionState({});
+  useEffect(() => {
+    form.batch(() => {
+      [
+        "contract",
```

Comment:
these key are defined in the PrimePrecarite tools. We can loop over `getState().values` key  to achieve the same thing and make it more general. That way we extract this step to `commons`  and use it in each tool.

Reply:
OK i'll do that